### ISSUE TITLE
Jquery UI 1.10 is deprecated and throws Lighthouse warning.

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#27186]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Jquery UI included in Prestashop is still on 1.10 (like 8 years old version).

It throws warning on category pages in Lighthouse, because it is considered a security risk. Faceted search uses it for price range slider.

![Výstřižek](https://user-images.githubusercontent.com/6097524/148055486-e0b90ce1-8d72-4398-aeec-223853f8b99d.JPG)

### Describe the solution you'd like

Several solutions I would like to do.
- Update this archaic library to current version - 1.13.
- Stop using it in BO by default. We can now use default html type=date and type=color inputs, with nice interfaces not needing any library.
- Use own lightweight solution in faceted search, so we don't have to load this library because of one slider. There is tons of it on google.

### Additional context

Ping @NeOMakinG

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by marwachelly** [comment]
Hello,

Thank you for your suggestion. @PrestaShop/prestashop-core-developers Could you please have a look?  
Is this https://github.com/PrestaShop/PrestaShop/issues/10463#issuecomment-425008014 still valid ?

Many thanks,

**Comment by Hlavtox** [comment]
@marwachelly jQuery has already been updated to 3.x.x in BO, core.js and classic theme. So I think we can use this oportunity to also update the UI plugin, when we have a next Major coming ;-)

**Comment by davidglezz** [comment]
I would like that we stop using it in BO and FO

**Comment by Hlavtox** [comment]
@mparvazi @NeOMakinG @Oksydan @SharakPL Guys, what do we do with this?

I would upgrade it in BO for v8 and remove it from faceted search somehow. WDYT?

**Comment by mparvazi** [comment]
> Use own lightweight solution in faceted search, so we don't have to load this library because of one slider.

IMO it's better solution.

**Comment by Hlavtox** [comment]
Yes, already finished in hummingbird, but how to handle the module itself? Do we add switch to it called "Load jQuery UI"?

**Comment by NeOMakinG** [comment]
If it's just a deprecation problem: we could update jQuery UI

If we want to get rid of it, why not

wdyt @PrestaShop/prestashop-maintainers ?

**Comment by matks** [comment]
@NeOMakinG Makes me think that as @Hlavtox is not in the group prestashop-maintainers he is not notified. 😅  something we need to fix

(we could create sub-teams)

So with the upgrade of jQuery, the problem is gone from FO on Classic theme, is that right? Which was the most annoying place for lighthouse

Now the problem remains in back-office (not a big issue if I am not mistaken) and on facetedsearch FO page (ah this one is a more frustrating issue, right?)

**Comment by Hlavtox** [comment]
@matks We already got rid of it (the usage of it) on hummingbird, it's not included by any other native module if I am not mistaken.

So, we can also remove the usage of it from classic.

But, the library will still be loaded by the module. :( And if we remove it from the module, we break themes that relied on it.

**Comment by NeOMakinG** [comment]
We definitely need a feature flag system on such things:

- Get rid of it behind a feature flag
- Let people migrate to the new Facetedsearch without jQuery UI and deprecate the jQuery UI version
- X month later, make the feature flag default and remove jQuery UI, bump major

That would be the smoothest solution, people would have a lot of time to migrate it 

Or bump to a new major version only compatible with 8.0 🤝

**Comment by Hlavtox** [comment]
@matks @NeOMakinG We need decision and action on this. Version 8 should not be released with a historic version of this library.

**Comment by matks** [comment]
In my opinion, Classic theme is full of outdated technologies so one more or one less is not a big issue IMO.

So I suggest we do the following: when Hummingbird is close to 1.0.0 final release, we update facetedsearch module and make it able to render differently whether it is being used with Classic Theme and Hummingbird. When facetedsearch module is executed on a shop where Hummingbird theme is used, we simply dont load jQuery UI.

Is that OK?

I think we did a nice job on pushing Classic Theme as far as we could in the future but at some point we got to admit Classic is the past and Hummingbird is the future.

**Comment by zuk3975** [comment]
ps_searchbar module probably needs same adjustments (as discussed in slack)

**Comment by Hlavtox** [comment]
@NeOMakinG reopen ;-)

**Comment by Hlavtox** [comment]
Still relevant

**Comment by boo-code** [comment]
Measured on `9.2.x`, since the technical part of this thread is from 2022 and one half of it has moved.

The vendored copy is unchanged: `js/jquery/ui/jquery-ui.js` still declares `jQuery UI - v1.10.3 - 2013-05-03`, and `js/jquery/ui/` is 50 files, 1.8 MB.

Two versions now coexist. The back office new theme pulls `jquery-ui-dist ^1.13.3` from npm, while legacy pages keep loading the 2013 bundle through `addJqueryUI()`. Six call sites remain in core:

- `classes/controller/AdminController.php:2686` - `ui.slider` and `ui.datepicker`
- `classes/helper/HelperList.php:720` and `classes/helper/HelperForm.php:215` - datepicker
- `controllers/admin/AdminCartRulesController.php:829`, `AdminDashboardController.php:26`, `AdminStatsTabController.php:71`

On the front office the situation is not the one the discussion above assumes.

`ps_facetedsearch` did get the switch from PrestaShop/ps_facetedsearch#691, but `PS_USE_JQUERY_UI_SLIDER` is written as `1` both on install and by `upgrade/upgrade-3.9.0.php`, and `src/Hook/ProductSearch.php:88` calls `addJqueryUi('ui.slider')` whenever it is truthy. So it is opt-out, and a default shop still loads the 2013 library on category pages.

`ps_searchbar` loads it too, unconditionally, in `hookDisplayHeader()`: `addJqueryUI('ui.autocomplete')`. That one cannot be switched off, because `ps_searchbar.js` extends the widget itself - `$.widget('prestashop.psBlockSearchAutocomplete', $.ui.autocomplete, {...})`. So a shop on a theme that avoids the slider still ships jQuery UI 1.10.3 to the front office as long as the search bar is enabled.

The remaining front-office blocker is therefore the search bar rather than faceted search, and the faceted-search default is a separate one-line decision.
